File: solutions/c/reverse-string/1/reverse_string.c
#include "reverse_string.h"
#include <stdlib.h>
#include <string.h>

char *reverse(const char *value)
{
   size_t value_length = strlen(value);
   char *result = malloc(value_length + 1);
   char *dest = result + value_length;

   *dest = '\0';
   while (*value) { *--dest = *value++; }
   return result;
}

File: solutions/javascript/reverse-string/1/reverse-string.js
//
// This is only a SKELETON file for the 'Reverse String' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const reverseString = (str) => {
  let result = "";
  for (let character of str) {
    result = character + result;
  }
  return result;
};
